app/routes.py

```diff
@@ -1,2 +1,269 @@
+                "details": "Invalid data"
+            }, 400)
+        else:
+            new_task = Task(title=request_body["title"],
+            description=request_body["description"],
+            completed_at=request_body["completed_at"])
+
+            db.session.add(new_task)
+            db.session.commit()
+
+            return make_response({
+                "task": {
+                    "id": new_task.task_id,
+                    "title": new_task.title,
+                    "description": new_task.description,
+                    "is_complete": new_task.is_complete()
+            }}, 201)
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        if task.goal_id is None:
+            return {
+                "task": {
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": task.is_complete()
+            }}
+        if task.task_id:
+            return {
+                "task": {
+                    "id": task.task_id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": task.is_complete()
+            }}
+        else:
+            return {
+                "task": {
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": task.is_complete()
+            }}
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.completed_at = form_data["completed_at"]
+
+        db.session.commit()
+        
+        headers = {"Authorization": os.environ.get("SLACK_KEY")}
+        data = {
+        "channel": "C021BV8A5V0", 
+        "text": f"Someone just completed the task {task.title}"
+        }
+        
+        requests.patch('https://slack.com/api/chat.postMessage',
+        headers=headers, data=data)
+
+
+        return {
+                "task": {
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": task.is_complete()
+            }}
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return {
+            "details": (f'Task {task.task_id} "{task.title}" successfully deleted')
+        }
+        
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
```

Comment:
Flask has a method called `.get_or_404()` that can refine these 3 lines into one! 
```suggestion
    task = Task.query.get_or_404(task_id)

```